src/meta_memcache/base/cache_pool.py

```diff
@@ -0,0 +1,251 @@
+            IntFlag.CACHE_TTL: ttl,
+        }
+        if cas is not None:
+            int_flags[IntFlag.CAS] = cas
+            if stale_policy and stale_policy.mark_stale_on_cas_mismatch:
+                flags.add(Flag.MARK_STALE)
+
+        result = self.ms(
+            key=key,
+            value=value,
+            ttl=ttl,
+            flags=flags,
+            int_flags=int_flags,
+        )
+
+        return isinstance(result, Success)
+
+    def delete(
+        self,
+        key: Key,
+        cas: Optional[int] = None,
+        no_reply: bool = False,
+        stale_policy: Optional[StalePolicy] = None,
+    ) -> bool:
+        flags = set()
+        int_flags = {}
+        if no_reply:
+            flags.add(Flag.NOREPLY)
+        if cas is not None:
+            int_flags[IntFlag.CAS] = cas
+        if stale_policy and stale_policy.mark_stale_on_deletion_ttl > 0:
+            flags.add(Flag.MARK_STALE)
+            int_flags[IntFlag.CACHE_TTL] = stale_policy.mark_stale_on_deletion_ttl
+
+        result = self.md(
+            key=key,
+            flags=flags,
+            int_flags=int_flags,
+        )
+
+        return isinstance(result, Success)
+
+    def touch(self, key: Key, ttl: int, no_reply: bool = False) -> bool:
+        flags = set()
+        int_flags = {IntFlag.CACHE_TTL: ttl}
+        if no_reply:
+            flags.add(Flag.NOREPLY)
+        result = self.mg(key, flags=flags, int_flags=int_flags)
+
+        return isinstance(result, Success)
+
+    # pyre-ignore[3]
+    def get_or_lease(
+        self,
+        key: Key,
+        lease_policy: LeasePolicy,
+        touch_ttl: Optional[int] = None,
+        recache_policy: Optional[RecachePolicy] = None,
+    ) -> Optional[Any]:
+        value, _ = self.get_or_lease_cas(
+            key=key,
+            lease_policy=lease_policy,
+            touch_ttl=touch_ttl,
+            recache_policy=recache_policy,
+        )
+        return value
+
+    # pyre-ignore[3]
+    def get_or_lease_cas(
+        self,
+        key: Key,
+        lease_policy: LeasePolicy,
+        touch_ttl: Optional[int] = None,
+        recache_policy: Optional[RecachePolicy] = None,
+    ) -> Tuple[Optional[Any], Optional[int]]:
+        if lease_policy.miss_retries <= 0:
+            raise ValueError(
+                "Wrong lease_policy: miss_retries needs to be greater than 0"
+            )
+        flags = {
+            Flag.RETURN_VALUE,
+            Flag.RETURN_TTL,
+            Flag.RETURN_CLIENT_FLAG,
+            Flag.RETURN_CAS,
+            Flag.RETURN_LAST_ACCESS,
+            Flag.RETURN_FETCHED,
+        }
+        int_flags = {
+            IntFlag.MISS_LEASE_TTL: lease_policy.ttl,
+        }
+        if recache_policy:
+            int_flags[IntFlag.RECACHE_TTL] = recache_policy.ttl
+        if touch_ttl is not None and touch_ttl >= 0:
+            int_flags[IntFlag.CACHE_TTL] = touch_ttl
+        i = 0
+        while True:
+            if i > 0:
+                print(
```

Comment:
Are all these prints for debugging purposes?

Reply:
yep, ops